File: frontend/src/services/MessagingService.js
import api from './api';

const MessagingService = {
  getContacts: async () => {
    const response = await api.get('/messaging/contacts');
    return response.data;
  },

  getChatHistory: async (otherUserId) => {
    const response = await api.get(`/messaging/history/${otherUserId}`);
    return response.data;
  },

  sendMessage: async (receiverId, content) => {
    const response = await api.post('/messaging/send', { receiverId, content });
    return response.data;
  },

  getUnreadCount: async () => {
    const response = await api.get('/messaging/unread');
    return response.data.length;
  }
};

export default MessagingService;

File: backend/src/main/java/com/obs/backend/service/MessagingService.java
package com.obs.backend.service;

import com.obs.backend.dto.student.ContactDTO;
import com.obs.backend.dto.student.MessageDTO;

import java.util.List;

public interface MessagingService {
    MessageDTO sendMessage(String senderUsername, Long receiverId, String content);
    List<MessageDTO> getChatHistory(String username, Long otherUserId);
    List<MessageDTO> getUnreadMessages(String username);
    List<ContactDTO> getContacts(String username);
    void markAsRead(Long messageId);
}
